operator/v1/types_csi_cluster_driver.go

```diff
@@ -96,6 +96,44 @@ type ClusterCSIDriverSpec struct {
 	// The current default behaviour is Managed.
 	// +optional
 	StorageClassState StorageClassStateName `json:"storageClassState,omitempty"`
+
+	// DriverConfig field can be used to optionally specify
+	// driver specific configuration.
+	// If this field is nil, platform chosen default configuration
+	// will be applied.
+	// +kubebuilder:validation:Optional
+	// +optional
+	DriverConfig *CSIDriverConfigSpec `json:"driverConfig,omitempty"`
```

Comment:
I was reading about API conventions and although I have accepted this suggestion, one of the things which I find sometimes problematic is - `CSIDriverConfigSpec` says `DriverName` is a *required* field and hence by that definition - an empty `CSIDriverConfigSpec` isn't a valid object because `DriverName` is not set.  I know on the wire, if field is empty - it is entirely omitted but I think that after deserializing we basically have a GO object that is essentially invalid and does not conform to scheme it defines itself, which is weird. 

So I think for union types which have a required union discriminator I think that a nil pointer is only sensible default if users don't want to specify one.

Reply:
It's a trade off we have decide to make for the discoverability benefits. But yes, you need to teach your controller to handle the empty `DriverConfig`. It shouldn't be too difficult and isn't dissimilar to the nil check you'd have to otherwise do
```
emptyConfig := operatorv1.CSIDriverConfigSpec{} 
if spec.DriverConfig == emptyConfig {
  // No driver config was specified
  return nil
}
```